GAE/test/org/akvo/flow/api/app/DataPointServletTest.java

```diff
@@ -0,0 +1,214 @@
+
+        dpa.setDeviceId(deviceId);
+        dpa.setDataPointIds(dataPointIds);
+        dpa.setSurveyAssignmentId(assignmentId);
+        dpa.setSurveyId(surveyId);
+
+        return dao.save(dpa);
+    }
+
+    private SurveyAssignment createAssignment(Long surveyId, List<Long> deviceIds, List<Long> formIds) {
+        final SurveyAssignment sa = new SurveyAssignment();
+        final SurveyAssignmentDao dao = new SurveyAssignmentDao();
+
+        sa.setSurveyId(surveyId);
+        sa.setDeviceIds(deviceIds);
+        sa.setFormIds(formIds);
+
+        return dao.save(sa);
+    }
+
+    private List<SurveyedLocale> createDataPoints(Long surveyId, int howMany) {
+        final SurveyedLocaleDao dpDao = new SurveyedLocaleDao();
+        final List<SurveyedLocale> datapoints = new ArrayList<>();
+        for (int i = 0; i < howMany; i++) {
+            SurveyedLocale dataPoint = new SurveyedLocale();
+            dataPoint.setIdentifier(String.valueOf(i));
+            dataPoint.setSurveyGroupId(surveyId);
+            dataPoint.setDisplayName("dataPoint: " + i);
+            datapoints.add(dataPoint);
+        }
+        return new ArrayList<>(dpDao.save(datapoints));
+    }
+
+    private Set<Long> getEntityIds(List<? extends BaseDomain> entities) {
+        Set<Long> entityIds = new HashSet<>();
+        for (BaseDomain o : entities) {
+            entityIds.add(o.getKey().getId());
+        }
+        return entityIds;
+    }
+
+    @Test
+    public void datastoreReadWriteTest() {
+        final Long assignmentId = 1L;
+        final Long deviceId = 1L;
+        final List<Long> dataPointIds = Collections.singletonList(0L);
+        final Long surveyId = 1L;
+
+        createDataPointAssignment(assignmentId, deviceId, dataPointIds, surveyId);
+
+        final DataPointAssignmentDao dataPointAssignmentDao = new DataPointAssignmentDao();
+        final List<DataPointAssignment> dataPointAssignments = dataPointAssignmentDao.listByDeviceAndSurvey(deviceId, surveyId);
+        assertFalse(dataPointAssignments.isEmpty());
+
+        final DataPointAssignment assignment = dataPointAssignments.get(0);
+        assertEquals(dataPointIds, assignment.getDataPointIds());
+        assertEquals(deviceId, assignment.getDeviceId());
+        assertEquals(surveyId, assignment.getSurveyId());
+
+
+        final List<Long> deviceIds = Arrays.asList(1L, 2L);
+        final List<Long> formIds = Arrays.asList(100L, 200L);
+
+        createAssignment(surveyId, deviceIds, formIds);
+
+        final SurveyAssignmentDao saDao = new SurveyAssignmentDao();
+        final List<SurveyAssignment> surveyAssignments = saDao.listAllContainingDevice(2L);
+        assertFalse(surveyAssignments.isEmpty());
+
+        final SurveyAssignment sa = surveyAssignments.get(0);
+        assertEquals(surveyId, sa.getSurveyId());
+        assertTrue(sa.getDeviceIds().contains(2L));
+        assertEquals(formIds, sa.getFormIds());
```

Comment:
Does this part of the test has anything to do with the previous part about `dataPointAssignmentDao`? If not, move to another tests

Reply:
These are two different tests. Please split in two.